Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/FAQFragment.kt

```diff
@@ -0,0 +1,198 @@
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        val view = inflater.inflate(R.layout.faq_fragment, container, false)
+        expandableListView = view.findViewById<View>(R.id.faqlist) as ExpandableListView
+        fetchDataFromFirebase()
+        return view
+    }
+
+    private fun fetchDataFromFirebase() {
+        firebaseHelper.faqDatabase.addListenerForSingleValueEvent(object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                val rows = ArrayList<FaqEvent>()
+                for (faqSnapshot in dataSnapshot.children) {
+                    val data = faqSnapshot.getValue(FaqEvent::class.java)
+                    if (data != null) {
+                        rows.add(data)
+                    }
+                }
+
+                expandableListView.setAdapter(FaqExpandable(rows.toList()))
+            }
+
+            override fun onCancelled(databaseError: DatabaseError) {
+                Log.e(javaClass.canonicalName, "onCancelled", databaseError.toException())
+            }
+        })
+
+    }
+
+    inner class FaqExpandable(faqData: List<FaqEvent>) : BaseExpandableListAdapter() {
+
+        val questionsGroup: List<String?>
+        val childrens: HashMap<Int, List<Answer>?>
+        init {
+            questionsGroup = faqData.map { it.question }
+            childrens = HashMap()
+            faqData.forEachIndexed {i, faq ->
+                childrens.put(i, faq.answers)
+            }
+        }
+
+        override fun getGroupCount(): Int {
+            return questionsGroup.size
+        }
+
+        override fun getChildrenCount(position: Int): Int {
+            return if (childrens[position] == null) 0 else childrens[position]!!.size
+        }
+
+        override fun getGroup(position: Int): List<Answer>? {
+            return childrens.get(position)
+        }
+
+        override fun getChild(groupId: Int, dex: Int): Answer? {
+            return childrens[groupId]?.get(dex)
+        }
+
+        override fun getGroupId(position: Int): Long {
+            return position.toLong()
+        }
+
+        override fun getChildId(groupPosition: Int, childPosition: Int): Long {
+            return (groupPosition * 100 + childPosition).toLong()
+        }
+
+        override fun hasStableIds(): Boolean {
+            return true
+        }
+
+        override fun getGroupView(groupPosition: Int, isExpanded: Boolean, convertView: View?, parent: ViewGroup): View {
+            var convertView = convertView
+            if (convertView == null) {
+                convertView = LayoutInflater.from(context).inflate(R.layout.faq_header, null)
+            }
+            (convertView!!.findViewById<View>(R.id.question_text) as TextView).text = questionsGroup!![groupPosition]
```

Comment:
probably don't need the convertView!!...should be non-null at this point.
And you can do findViewById<TextView> to implicit cast

Reply:
needs !!